flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolImpl.java

```diff
@@ -447,12 +452,40 @@ public void releaseSlot(@Nonnull SlotRequestId slotRequestId, @Nullable Throwabl
 
 		componentMainThreadExecutor.assertRunningInMainThread();
 
+		checkState(batchSlotTimeoutCheckEnabled, "batch slot timeout check is disabled unexpectedly.");
+
 		final PendingRequest pendingRequest = PendingRequest.createBatchRequest(slotRequestId, resourceProfile);
 
 		return requestNewAllocatedSlotInternal(pendingRequest)
 			.thenApply(Function.identity());
 	}
 
+	@Override
+	public CompletableFuture<PhysicalSlot> requestNewAllocatedSlotWithoutTimeout(
```

Comment:
I am wondering whether it would be more clear just to add  `disableBatchSlotTimeoutCheck` explicitly to the public interface.
We could also make `timeout` `Nullable` in `requestNewAllocatedSlot` then `FutureUtils.orTimeout` can be conditional in `requestNewAllocatedSlot`.
The we do not need `requestNewAllocatedSlotWithoutTimeout` and make any assumptions in it about `batchSlotTimeoutCheckEnabled`.

Reply:
Agreed it would be better to explicitly disable the timeout check.

Reply:
done via 9b5ca3d921855ddc3557d72202526cec6c474be8 and 54bd4613db64de1ff5b7fcdf29931a54895c3163

Reply:
I do not see where we call `disableBatchSlotTimeoutCheck` now.
Should we add it to `BulkSlotProviderImpl#start`?
Also a test?

Reply:
Yeah my stupid mistake.
Will call it in `BulkSlotProviderImpl#start` and add a test

Reply:
done.